frontend/src/App.vue

```diff
@@ -63,18 +63,17 @@ export default {
     },
 
     getItemByFqid(fqid) {
-      let items = [];
+      let rootLevelItems = [];
       this.collections.forEach(col => {
-        items = items.concat(col.items);
+        rootLevelItems = rootLevelItems.concat(col.items);
       });
       const flatten = [];
-      let itemnode;
-      while (items.length > 0) {
-        itemnode = items.shift();
-        flatten.push(itemnode);
-        if (!itemnode.items) continue;
-        itemnode.items.forEach(item => {
-          items.push(item);
+      while (rootLevelItems.length > 0) {
+        let itemNode = rootLevelItems.shift();
+        flatten.push(itemNode);
+        if (!itemNode.items) continue;
```

Comment:
maybe we can use a stronger inspection and avoid the **continue** statement via:
```javascript
if (Array.isArray(itemNode.items) {
        itemNode.items.forEach(item => {
          items.push(item);
        });
}
```